### PR TITLE
Document frame_limiter module

### PR DESCRIPTION
**Pull request by randomPoison**
This PR adds documentation for the `frame_limiter` module and its members. I've tried to accurately describe the frame limiting behavior and its variations, though there are a few things I'm not certain I have right:

* I state that amethyst will attempt to keep a set cadence for starting frames based on the frame rate, but that's not quite accurate to the current implementation. The actual implementation guarantees that the next frame will start *no sooner* than `frame_duration` after the current frame, which will result in the frame  starts drifting over time. I assume the goal is to keep an actual cadence, and that the current implementation is either a bug or is somehow fixed in practice by vsync, but I might have documented the behavior incorrectly.
* I attempted to summarize the advantages/drawbacks of the different frame limiting approaches as I understand them, though I don't know if in practice they limiting behaviors currently work as described (e.g. I've still observed relatively high CPU usage with the `Sleep` strategy, may be related to #780).

---
This change is [Reviewable](https://reviewable.io/reviews/amethyst/amethyst/783)

### CODE REVIEW

File: amethyst_core/src/frame_limiter.rs [hunk 1]
```diff
@@ -1,22 +1,118 @@
...
+//!
+//! # Frame Rate Limiting Strategies
+//!
+//! The four possible strategies described by [`FrameRateLimitStrategy`] are as follows:
+//!
+//! * `Unlimited` will not try to limit the frame rate to the specified maximum. Amethyst
+//!   will call [`thread::yield_now`] once and then continue to the next frame.
+//! * `Yield` will call [`thread::yield_now`] repeatedly until the frame duration has
+//!   passed. This will result in the most accurate frame timings, but effectively guarantees
+//!   that one CPU core will be fully utilized during the frame's idle time.
```
[the remark this answers is not in the document]
Comment (randomPoison) [reply amethyst_core/src/frame_limiter.rs]: Even if the system gives the core to another thread, the core will still be active while running that . So either we yield the core to another thread if one is available, or the system keeps our thread running. Either way, it seems like at least one core is guaranteed to be active and running at 100% as long as we're spin-waiting using `yield_now`. Does that seem accurate to you, or am I missing something?

**Comment by randomPoison** [comment]
Should any of the documentation change to reflect that precision timing isn't currently a goal? Maybe the documentation was wishful thinking on my part, but I'd personally like to see amethyst handle frame timing in such a way that frames have a reliable cadence and duration.

**Comment by Rhuagh** [comment]
I'd say we should document the current behavior, and create an issue or PR for improving it. The frame limiter was kind of thrown together in like 10 minutes because we needed some kind of limiter, it's not perfect in any way :)

File: amethyst_core/src/frame_limiter.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Rhuagh) [reply amethyst_core/src/frame_limiter.rs]: Are threads guaranteed to yield if they are waiting for locks? I don't believe they are, depends on the lock. If so, you could theoretically have virtually no cpu usage while spin waiting. But I digress, leave the current formulation, it's accurate for most scenarios.

**Comment by randomPoison** [comment]
I removed the discussion of frame cadence to avoid misrepresenting amethyst's current behavior.

File: amethyst_core/src/frame_limiter.rs [hunk 2]
```diff
@@ -1,22 +1,98 @@
...
+//! frame completed sufficiently quickly.
+//!
+//! The frame rate limiting strategy has two parts: A maximum frame rate, given as a number of
+//! frames per second, and a strategy for returning any remaining time in the frame to the
+//! operating system. Based on the specified maximum frame rate, each frame has a budget for
+//! how long it can take. For example, at 60 fps each frame has 16.6 milliseconds to perform
+//! any work it needs to. If a frame takes less time than is budgeted, amethyst will attempt to
+//! yield the remaining time back to the operating system, using the chosen strategy.
+//!
+//! By default, the amethyst will set the maximum frame rate to 144 fps, and will use a yield-only
```
Comment (torkleyy) [inline amethyst_core/src/frame_limiter.rs]: s/the amethyst/Amethyst

**Pull request merged**